ui/component/page/view.jsx

```diff
@@ -121,33 +121,41 @@ function Page(props: Props) {
       >
         {getSideNavElem()}
 
-        <main
-          id={'main-content'}
-          className={classnames(MAIN_CLASS, className, {
-            'main--full-width': fullWidthPage,
-            'main--auth-page': authPage,
-            'main--file-page': filePage,
-            'main--settings-page': settingsPage,
-            'main--markdown': isMarkdown,
-            'main--theater-mode': isOnFilePage && videoTheaterMode && !livestream,
-            'main--livestream': livestream && !chatDisabled,
+        <div
+          className={classnames({
+            'sidebar--pusher': fullWidthPage,
+            'sidebar--pusher--open': sidebarOpen && fullWidthPage,
+            'sidebar--pusher--filepage': !fullWidthPage,
           })}
         >
-          {children}
+          <main
+            id={'main-content'}
+            className={classnames(MAIN_CLASS, className, {
+              'main--full-width': fullWidthPage,
+              'main--auth-page': authPage,
+              'main--file-page': filePage,
+              'main--settings-page': settingsPage,
+              'main--markdown': isMarkdown,
+              'main--theater-mode': isOnFilePage && videoTheaterMode && !livestream,
+              'main--livestream': livestream && !chatDisabled,
+            })}
+          >
+            {children}
 
-          {!isMobile && rightSide && <div className="main__right-side">{rightSide}</div>}
-        </main>
+            {!isMobile && rightSide && <div className="main__right-side">{rightSide}</div>}
+          </main>
+          {/* @if TARGET='web' */}
+          {!noFooter && (
+            <React.Suspense fallback={null}>
+              <Footer />
```

Comment:
Just a note to other reviewers:  moving the footer inside does make the footer closer to the last element.  When signing in from mobile, for example, the footer is closer to the main element.

Should be fine, I think.

Reply:
If this isn't acceptable, that is ok. It will probably just take more time to fix. I want to make sure this PR is perfect before merging since it's changing a lot of the core page structure.

The reason I moved the footer was because I made the sidebar pusher positioned absolutely.
So if the footer is outside the sidebar-pusher it looks like this:
![image](https://user-images.githubusercontent.com/7473983/145886705-d3527ca6-7d3a-4578-bf5d-40f076f26099.png)

I will take a look at fixing those other issues. 

I can probably fix the footer being closer to the main element by adding a minimum height on the main element or something like that.

Can you show me how to get to the cumulative layout shift scores? It was probably something I didn't test.

Reply:
It's available under "Lighthouse" in devtools

Before and after
<img src="https://user-images.githubusercontent.com/64950861/145914307-7c4cbb06-4a2d-4106-b27a-fa4b641a4ee1.png" width="200">        <img src="https://user-images.githubusercontent.com/64950861/145914343-27e0e317-87d5-4897-87a1-71ddc87e4118.png" width="200">

The tool sometimes show inverted results (various factors can affect it).  Since we can visually see the shift for this case, we probably don't need to the tool to verify.

Reply:
Ah ok I understand now. Yeah I agree it's a problem, and from what I read it is probably affecting the cumulative layout shift scores. I will spend some more time resolving this. I think I have to move the header inside the sidebar pusher, but there should be a way to style it so it's at the bottom when there's extra space. It's just hard for me to test, since I am having trouble mocking the loading state. I just don't know the layout of the app very well. I will get back to you once I make some more progress.

Reply:
@infinite-persistence I added a minimum height to the main content. This should push the footer to the bottom when there isn't enough content. It should be very close to what is in master.